src/reconcile/rec_write.c

```diff
@@ -2207,7 +2207,8 @@ __rec_write_wrapup(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_PAGE *page)
              */
         if (F_ISSET(r, WT_REC_IN_MEMORY) || r->multi->supd_restore) {
             WT_ASSERT(session, F_ISSET(r, WT_REC_IN_MEMORY) ||
-                (F_ISSET(r, WT_REC_EVICT) && r->leave_dirty && r->multi->supd_entries != 0));
+                (F_ISSET(r, WT_REC_EVICT) && (r->leave_dirty || !F_ISSET(r, WT_REC_HS)) &&
```

Comment:
The change looks fine, but can you add an additional check to confirm that it is a fixed column store like in `__rec_need_save_upd` function.

Reply:
yeah, I can make it more specific.